frontend/lexer.py

```diff
@@ -226,6 +226,60 @@ def LookPastSpace(self, lex_mode):
 
         return tok_type
 
+    def LookAheadDParens(self, shift_back):
+        # type: (int) -> bool
+        """For finding the closing arithmetic parens - ))
+        If none consecutive parens (not separated by any other token) occur,
+        returns False.
+
+        shift_back is the number of characters to go back for in case the
+        expression is not arithmetic.
+        """
+        original_pos = self.line_pos
+        tok_type, end_pos = match.OneToken(lex_mode_e.Arith,
```

Comment:
Style: I personally prefer having a `while True ` loop with  single `match.OneToken()` call, with an `if ... break` for the condition

Since it is the same way to advance in both cases